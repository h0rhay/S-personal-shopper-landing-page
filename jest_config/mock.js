export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
