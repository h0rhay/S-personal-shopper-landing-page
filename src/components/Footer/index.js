import React from 'react';
import styled, { css } from 'styled-components';
import SelfridgesLogo from '../../images/selfridges_logo.svg';
import SiteWidthWrapper from '../SiteWidthWrapper';

const PageFooter = styled.footer`
  height: 5.625rem;
  padding: 2rem 0;
`;

const SiteLogo = styled(SelfridgesLogo)`
  width: 8rem;
  flex-grow: 1;
  path {
    fill: ${({ theme }) => theme.text};
  }
  ${({ theme }) => css`
    @media ${theme.breakpoints.small} {
      margin: 2rem 0;
    }
  `}
`;

const Container = styled.div`
  display: flex;

  ${({ theme }) => css`
    @media ${theme.breakpoints.small} {
      flex-direction: column;
    }
  `}
`;

const NavBar = styled.ul`
  flex-grow: 9;
  justify-content: end;
  display: flex;

  li {
    display: inline;
  }

  ${({ theme }) => css`
    @media ${theme.breakpoints.small} {
      flex-direction: column;
      li:last-of-type {
        margin-bottom: 2rem;
      }
    }
    @media ${theme.breakpoints.medium} {
      li + li {
        &::before {
          content: 'â€¢';
          margin: 0 0.5rem;
        }
      }
    }
  `};
`;

const FooterLinks = styled.a`
  font-family: 'DINNextLTPro-Light';
  font-size: 14px;
  margin: 0 5px;

  ${({ theme }) => css`
    @media ${theme.breakpoints.small} {
      margin-bottom: 0.5rem;
    }
  `};
`;

const NavLink = styled.a`
  span {
    display: none;
  }
`;

const Footer = () => (
  <PageFooter>
    <SiteWidthWrapper>
      <Container>
        <NavLink
          data-testid="selfridges-footer-logo"
          href="https://www.selfridges.com/"
          target="_blank"
          rel="noopener noreferrer"
          title="Selfridges &amp; Co"
        >
          <SiteLogo />
          <span>Selfridges &amp; Co</span>
        </NavLink>
        <NavBar>
          <li>
            <FooterLinks href="https://www.selfridges.com/GB/en/features/info/accessibility">ACCESSIBILITY</FooterLinks>
          </li>
          <li>
            <FooterLinks href="https://www.selfridges.com/GB/en/features/info/our-corporate-policies/privacy-cookie-policy">
              PRIVACY & COOKIES POLICIES
            </FooterLinks>
          </li>
          <li>
            <FooterLinks href="https://www.selfridges.com/GB/en/features/info/terms-conditions">
              TERMS & CONDITIONS
            </FooterLinks>
          </li>
        </NavBar>
      </Container>
    </SiteWidthWrapper>
  </PageFooter>
);

export default Footer;
