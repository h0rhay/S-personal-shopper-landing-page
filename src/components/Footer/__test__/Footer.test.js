import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import Footer from '../index';

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <Footer />
    </ThemeProvider>
  );
let container;
describe('Footer', () => {
  beforeEach(() => {
    container = renderComponent({ theme: mockTheme }).container;
  });
  it('should match screenshot', () => {
    expect(container).toMatchSnapshot();
  });
});
