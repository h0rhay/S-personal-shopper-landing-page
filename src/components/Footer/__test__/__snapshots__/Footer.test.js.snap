// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match screenshot 1`] = `
<div>
  <footer
    class="sc-gsDKAQ eYhrHw"
  >
    <div
      class="sc-bdvvtL kiJDep site_width_wrap"
    >
      <div
        class="sc-hKwDye xtTyt"
      >
        <a
          class="sc-gKclnd kpKMdZ"
          data-testid="selfridges-footer-logo"
          href="https://www.selfridges.com/"
          rel="noopener noreferrer"
          target="_blank"
          title="Selfridges & Co"
        >
          <test-file-stub
            classname="sc-dkPtRN imBqKc"
          />
          <span>
            Selfridges & Co
          </span>
        </a>
        <ul
          class="sc-eCImPb egwiBT"
        >
          <li>
            <a
              class="sc-jRQBWg cgTkOP"
              href="https://www.selfridges.com/GB/en/features/info/accessibility"
            >
              ACCESSIBILITY
            </a>
          </li>
          <li>
            <a
              class="sc-jRQBWg cgTkOP"
              href="https://www.selfridges.com/GB/en/features/info/our-corporate-policies/privacy-cookie-policy"
            >
              PRIVACY & COOKIES POLICIES
            </a>
          </li>
          <li>
            <a
              class="sc-jRQBWg cgTkOP"
              href="https://www.selfridges.com/GB/en/features/info/terms-conditions"
            >
              TERMS & CONDITIONS
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</div>
`;
